### PR TITLE
Fix recovery action collision check.

### PR DESCRIPTION
**Pull request by acyen**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #2189 |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | Turtlebot simulation |

---

## Description of contribution in a few bullet points

Fix collision check in `back_up.cpp` and also `spin.cpp`.

## Description of documentation updates required from your changes
None

### CODE REVIEW

File: nav2_recoveries/plugins/back_up.cpp [hunk 1]
```diff
@@ -120,11 +120,12 @@ bool BackUp::isCollisionFree(
   double sim_position_change;
   const double diff_dist = abs(command_x_) - distance;
   const int max_cycle_count = static_cast<int>(cycle_frequency_ * simulate_ahead_time_);
-
+  geometry_msgs::msg::Pose2D init_pose = pose2d;
```
Comment (SteveMacenski) [inline nav2_recoveries/plugins/back_up.cpp:123]: Did you test this to fix your issue with too long looking back?

**Comment by SteveMacenski** [comment]
@acyen I'm happy to also merge this into `Foxy`, but please also direct a PR to `main` so that this change is reflected into all newer distributions for now on and not just in `Foxy`.

File: nav2_recoveries/plugins/back_up.cpp [hunk 2]
```diff
@@ -120,11 +120,12 @@ bool BackUp::isCollisionFree(
...
+
```
Comment (SteveMacenski) [inline nav2_recoveries/plugins/back_up.cpp:124]: 12:48:51 3: /tmp/ws/src/navigation2/nav2_recoveries/plugins/back_up.cpp:124:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]

Linting issue as well

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: nav2_recoveries/plugins/back_up.cpp (hunk 1 above)
Comment (acyen) [reply nav2_recoveries/plugins/back_up.cpp:123]: Yes, manual test using tb3 simulation.

**Review by SteveMacenski — APPROVED** [review]
@acyen please submit a PR with this change to `main` as well so that its reflected in all new distributions as well, not just Foxy

**Pull request merged**